### DISCUSSION TITLE
Pull Request에 스터디 주차 표시 방법

### DISCUSSION THREAD

**Discussion by DaleSeo** [DaleStudy/leetcode-study#71] [공지사항]
그동안 스터디 멤버들이 각자 다른 방법으로 Pull Request에 몇 주차 스터디에 대한 답안 제출인지를 표시해주셨는데요.
이로 인해, GitHub Projects를 통해서 스터디의 전체적인 진행 상황을 파악하고 공유드리는데 계속해서 혼선이 발생하였습니다.

제가 지금까지 생성해주신 Pull Reqest를 살펴보니, 크게 다음 3가지 방법으로 스터디 주차가 표시되어 있더라고요.

- 레이블(label) 달기
- 마일스톤(miliestone) 설정
- 이터레이션(iteration) 설정

지금부터는 이터레이션(iteration) 설정을 통해서 Pull Request 스터디 주차를 표시하는 것으로 통일하려고 합니다.
이터레이션은 프로젝트의 필드이기 때문에 자동으로 생성될 뿐만 아니라 그룹핑, 필터링, 정렬을 하기가 매우 용이하게 때문입니다.
Pull Request에 이터레이션을 설정하는 방법은 아래 스크린샷을 참고 바랍니다.

![Shot 2024-05-12 at 14 07 13@2x](https://github.com/DaleStudy/leetcode-study/assets/5466341/b75d6397-5cdf-4491-8d7a-8b0564c6648c)

> [!Warning]
> 스터디 답안 제출이 아닌 일반적인 저장소 개선을 위한 Pull Request를 올려주실 때는 이터레이션 설정할 필요 없습니다.

**Comment by DaleSeo** [comment]
Pull Requeset 올려주시는 날짜/시간에 따라서 이터레이션(iteration) 설정을 자동화해보려고 했었는데, 예상했던 것보다 고려해야할 부분이 많더라고요. 문제를 공지하자 마자 바로 풀어서 PR를 올려주시는 분들도 있고, 마감보다 하루 이틀 늦게 PR를 올려주시는 분까지도 감안을 하면, 조금 귀찮더라도 PR 작성자가 직접 설정해주는 것이 더 낫다고 판단했습니다. 대안으로 PR에 레이블(label)이나 마일스톤(milestone)을 설정하면, 자동으로 이터레이션(iteration)을 설정해주는 것인데, 어차피 레이블/마일스톤 설정하는거나 이터레이션을 설정하는거나 노력은 거기서 거기인 것 같아서요. 의미없는 자동화라는 생각이 들었습니다. 또 혼선이 발생할 수도 있고요. 혹시 관련해서 다른 좋은 자동화 아이디어 있으시면 댓글로 알려주세요.

**Reply by SamTheKorean** [reply]
자동화를 시키면 예정보다 조금 늦게 제출하시는 분들도 고려해야 해서 번거롭겠군요. 개인적으로 이정도 설정은 직접 해주면 확인 하면서 체크하는 느낌이라 성취감도 들고 좋을 것 같습니다ㅎㅎ

**Reply by sounmind** [reply]
안녕하세요! 만약 주차별로 풀어야 할 문제는 고정되어 있으므로 ...
해당 주차 알고리즘 문제 이름의 디렉토리 하위의 변경사항이 담긴 PR이 OPEN되면 
그 주차에 맞는 이터레이션을 할당하도록 workflow를 작성할 수도 있지 않을까, 생각했습니다.

이전 회사에서 비슷한 workflow를 작성한 적 있는데, 한 번 시도하고 PR을 올려봐도 될까요?!

**Reply by DaleSeo** [reply]
@sounmind 네, 그럼요! 😄
